### ISSUE TITLE
DigitalExpression empty output

### ISSUE THREAD

**Issue by pavsol** [broadinstitute/Drop-seq#180]
Hello,
I am running drop-seq pipeline (version 2.3.0) using provided scripts. A sucesfully passed the alignment step using Drop-seq_alignment.sh. Then, I tried to run DigitalExpression.sh. The process seemed to be running correctly but it gave me an empty output file containing only the first row with the cell tags. The summary file was not generated at all.
When started the script, following warnings appeared:

```
12:02:57.745 WARN  NativeLibraryLoader - Unable to load libgkl_compression.so from native/libgkl_compression.so (No such file or directory)
12:02:57.806 WARN  IntelDeflaterFactory - IntelInflater is not supported, using Java.util.zip.Inflater
```
But obviously, there is libgkl_compression.so in the jar file:
```
jar tf Drop-seq_tools-2.3.0/3rdParty/picard/picard.jar | grep "compression.so"
com/intel/gkl/native/libgkl_compression.so
```

Mentioned warning appeared even when I run upstream scripts but everything finished without any errors. Do you have any idea what is going wrong?
Thank you.

**INPUT BAM:**
```
samtools view ~/Desktop/single/alignment_output/SRR8206654/final.bam | head -n 2
SRR8206654.14458821	16	Chr1	3479	255	64M	*	0	0	ACCTTCTATTGTTTTAATAATCTTTATAATTGTTAGGACCGGTAAATCTATTCCAATTAAACCG	-7--F7--AJ<JJF7--JFAJFJJJ7---FF7JF77-7-<-AF7-----7A-77--JF<7----	XC:Z:TTGTGGAGCCCA	MD:Z:3A2A19A19A1C2A12	XF:Z:INTERGENIC	PG:Z:STAR	RG:Z:A	NH:i:1	NM:i:6	XM:Z:GGACACAA	ZP:i:65	UQ:i:72	AS:i:50
SRR8206654.68939670	0	Chr1	3627	255	84M8S	*	0	0	CTCCAAATTATTAGATATACCAAACCAGAGAAAACAAATACATAACCGGTGCAATACAGATCACAGCGAGCGCGGGCGCTCCACCGCGCAGC	77-7A7<7--<<F-JJJ-7FA7JA<<7--7A-A7<--FA---F-<-77<-A--7<7F---<--7-7-----7-A-7-<--7-----7---)-	XC:Z:TATCCCCAGCTT	MD:Z:45T3A1A9T4A5A1A1A1A2G2	XF:Z:UTR	PG:Z:STAR	RG:Z:A	XG:Z:AT1G01010	NH:i:1	NM:i:10	XM:Z:TAGCGAAT	UQ:i:120	AS:i:62	ZS:i:9	gf:Z:UTR,INTERGENIC	gn:Z:AT1G01010,AT1G01010	gs:Z:+,+
```

**COMMAND:**
```./Drop-seq_tools-2.3.0/DigitalExpression I=alignment_output/SRR8206654/final.bam O=DGE/SRR8206654.dge.txt SUMMARY=SRR8206654.summary.txt NUM_CORE_BARCODES=100```

**OUTPUT FILE:**
GENE	TATCTGATTGGC	GAATTACCTCTT	TCGGAGCCCTGA ... (up to 100 barcodes)

**Comment by jamesnemesh** [comment]
Hi,

I don’t see anything clearly wrong with either your command or the bit of the BAM you submitted.   I assume you see something in the log that looks like:

INFO	2020-03-04 09:49:43	BarcodeListRetrieval	Looking for the top 100 cell barcodes
INFO	2020-03-04 09:49:43	BarcodeListRetrieval	Selected 100 core barcodes
INFO	2020-03-04 09:49:43	DigitalExpression	Calculating digital expression for [100] cells.

If you don’t see that, that would be good to know.

I’m not sure how big the BAM is, but perhaps you can share a larger subset of the data so I could take a look - either a single config like chromosome 22, or a downsampled version of the data.

-Jim

**Comment by pavsol** [comment]
Hi Jim,
thank you greatly for the fast response!
As usual, I figured out the problem just after I posted this. I had the same issue as in #73 
`Exception in thread "main" htsjdk.samtools.util.RuntimeIOException: java.nio.file.NoSuchFileException`
I created tmp folder in my working directory and added `TMP_DIR=tmp` to my command. So far, everything seem working well and I am obtaining results.

Thank you again.
Pavel
